src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+								logEventInfo.lastSeenVersion = lastSeenVersion;
+								logEventInfo.numSuccessfulClips = numClips;
+
+								let badRatingTimingDelayIsOver: boolean = RatingsHelper.badRatingTimingDelayIsOver(lastBadRatingDate, Date.now());
+								let badRatingVersionDelayIsOver: boolean = RatingsHelper.badRatingVersionDelayIsOver(lastBadRatingVersion, lastSeenVersion);
+								let clipSuccessDelayIsOver: boolean = RatingsHelper.clipSuccessDelayIsOver(numClips);
+
+								logEventInfo.badRatingTimingDelayIsOver = badRatingTimingDelayIsOver;
+								logEventInfo.badRatingVersionDelayIsOver = badRatingVersionDelayIsOver;
+								logEventInfo.clipSuccessDelayIsOver = clipSuccessDelayIsOver;
+
+								if (badRatingTimingDelayIsOver && badRatingVersionDelayIsOver && clipSuccessDelayIsOver) {
+									return resolve(true);
+								}
+
+								return resolve(false);
+							});
+						});
+					});
+				});
+			});
+		});
+	}
+
+	private static combineClientTypeAndSuffix(clientType: ClientType, suffix: string): string {
+		return ClientType[clientType] + suffix;
+	}
+
+	/**
+	 * Based on the three-part version number convention "X.Y.Z", where:
+	 *   * X = major version
+	 *   * Y = minor version
+	 *   * Z = patch
+	 * Returns the major version, X
+	 */
+	private static getMajorVersion(versionNum: string): number {
+		if (!Utils.isNullOrUndefined(versionNum)) {
+			let versionSplit: string[] = versionNum.split(".");
+
+			if (!Utils.isNullOrUndefined(versionSplit)) {
+				return parseInt(versionSplit[0], 10);
+			}
+		}
+	}
+
+	/**
+	 * Based on the three-part version number convention "X.Y.Z", where:
+	 *   * X = major version
+	 *   * Y = minor version
+	 *   * Z = patch
+	 * Returns the minor version, Y
+	 */
+	private static getMinorVersion(versionNum: string): number {
+		if (!Utils.isNullOrUndefined(versionNum)) {
+			let versionSplit: string[] = versionNum.split(".");
+
+			if (!Utils.isNullOrUndefined(versionSplit)) {
+				return parseInt(versionSplit[1], 10);
+			}
+		}
+	}
+
+	private static getRateUrlSettingNameForClient(clientType: ClientType): string {
+		return RatingsHelper.combineClientTypeAndSuffix(clientType, RatingsHelper.rateUrlSettingNameSuffix);
+	}
+
+	private static getRatingsPromptEnabledSettingNameForClient(clientType: ClientType): string {
+		return RatingsHelper.combineClientTypeAndSuffix(clientType, RatingsHelper.ratingsPromptEnabledSettingNameSuffix);
+	}
+
+	private static isValidDate(date: number): boolean {
+		let minimumTimeValue: number = (Constants.Settings.maximumJSTimeValue * -1);
+		return date >= minimumTimeValue && date <= Constants.Settings.maximumJSTimeValue;
+	}
+
+	private static setDoNotPromptStatus(): void {
+		Clipper.Storage.setValue(Constants.StorageKeys.doNotPromptRatings, "true");
+
+		Clipper.logger.logEvent(new Log.Event.BaseEvent(Log.Event.Label.SetDoNotPromptRatings));
+	}
+
+	private static versionHasCorrectFormat(version: string): boolean {
```

Comment:
As above with @VishaalK's comment, I think this logic already exists in version.ts, if not it can go in that file, or in another file in the versioning folder as appropriate.